Feature/modify ci cd

erges699 opened the pull request:
Добили с @Esedess  на ЕГО выделенном сервере!
Деплой проходит.
Осталось допилить копирование папки infra на удаленный сервер.
Но у Никиты заканчивается гранд через два дня! Так что надо бы PR смержить и крутить на боевом сервере, который отвалился.

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update Dockerfile

erges699 committed:
Delete Dockerfile

erges699 committed:
Create Dockerfile

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update Dockerfile

erges699 committed:
Update development.txt

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

kr0t commented on .github/workflows/stage_deploy.yaml:
```diff
@@ -29,34 +34,26 @@ jobs:
+        uses: docker/login-action@v2
         with:
-          registry: ${{ env.REGISTRY }}
+          registry: ghcr.io
```
У тебя определена переменная в блоке `env`, надо использовать её.

kr0t replied:
@erges699 ? Может стоит использовать переменную тут?

erges699 committed:
Update stage_deploy.yaml

SSH_PRIVATE_KEY

kr0t commented on .github/workflows/stage_deploy.yaml:
```diff
@@ -69,27 +66,15 @@ jobs:
           username: ${{ secrets.USERNAME }}
           key: ${{ secrets.SSH_PRIVATE_KEY }}
-          passphrase: ${{ secrets.PASSPHRASE }}
+          port: ${{ secrets.PORT }}        
```
Мы не меняли ssh порт, по дефолту у этого экшена 22, эта переменная тут не обязательна.
А почему ты решил убрать `PASSPHRASE`?

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

SSH_PUBLIC_KEY

kr0t commented on .github/workflows/stage_deploy.yaml:
```diff
@@ -69,27 +66,15 @@ jobs:
+
       - name: executing remote ssh commands to deploy
-        uses: appleboy/ssh-action@master
+        uses: appleboy/ssh-action@v0.1.7
```
А чем версия из `master` не подошла?

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

SSH_PRIVATE_KEY

kr0t reviewed (commented):
А где часть с копированием docker-compose на сервер и запуском приложения?

erges699 committed:
Update stage_deploy.yaml

SSH_PUBLIC_KEY

erges699 committed:
Update stage_deploy.yaml

fifsky/ssh-action@master

erges699 committed:
Update stage_deploy.yaml

- name: ls -a via ssh
  uses: fifsky/ssh-action@master
  with:
    command: |
      cd /tmp
      ls -a
    host: ${{ secrets.HOST }}
    user: root
    key: ${{ secrets.PRIVATE_KEY}}

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update Dockerfile

erges699 committed:
Create docker-compose-stage.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update Dockerfile

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update Dockerfile

erges699 committed:
Update Dockerfile

erges699 committed:
Update Dockerfile

erges699 committed:
Update Dockerfile

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update docker-compose-stage.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update settings.py

erges699 committed:
Update settings.py

erges699 committed:
Update settings.py

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update development.txt

erges699 committed:
Update development.txt

erges699 committed:
Update development.txt

erges699 committed:
Update development.txt

erges699 committed:
Update development.txt

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
docker image

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update development.txt

erges699 commented:
Конфликт решается:
#8 3.768 The conflict is caused by:
#8 3.768     The user requested platformdirs==3.0.0
#8 3.768     black 22.12.0 depends on platformdirs>=2
#8 3.768     virtualenv 20.17.1 depends on platformdirs<3 and >=2.4
#8 3.768 
#8 3.768 To fix this you could try to:
#8 3.768 1. loosen the range of package versions you've specified
#8 3.768 2. remove package versions to allow pip attempt to solve the dependency conflict

то есть если убрать версию у пакетов black и virtualenv

erges699 committed:
Update development.txt

erges699 committed:
Update development.txt

erges699 committed:
Update stage_deploy.yaml

kr0t commented on .github/workflows/stage_deploy.yaml:
```diff
@@ -14,8 +18,8 @@ on:
-  IMAGE_NAME: mssp_bot
-  DEPLOY_PATH: /MSPP
+  IMAGE_NAME: mssp
+  DEPLOY_PATH: "/MSPP"
```
Работает без кавычек.

erges699 replied:
Исправил

kr0t commented on .github/workflows/stage_deploy.yaml:
```diff
@@ -69,27 +65,48 @@ jobs:
+          target: ${{ env.DEPLOY_PATH }}
       - name: executing remote ssh commands to deploy
-        uses: appleboy/ssh-action@master
+        uses: appleboy/ssh-action@@master
```
Опечатка?

erges699 replied:
Переменную вставил.
Тут да, очепятался сегодня, когда редактировал

kr0t commented on src/config/settings.py:
```diff
@@ -9,14 +9,14 @@
-    dotenv_path = Path(BASE_DIR.parent / ".env")
-else:
-    dotenv_path = Path(BASE_DIR.parent / ".env_local")
+# if os.path.exists(BASE_DIR.parent / ".env"):
```
Ненужные комменты надо убрать.

erges699 replied:
Исправил (вернул в исходное)

kr0t commented on infra/docker-compose-stage.yml:
```diff
@@ -0,0 +1,32 @@
+      - ./.env-stage
+
+  nginx:
+    image: nginx:latest
```
Давай явно укажем версию nginx, можно взять любую одну из последних.
На тот случай, если появятся изменения, которые нам могут поломать функционал.

erges699 replied:
Исправил: 
image: nginx:1.23.3
Последняя версия: https://hub.docker.com/_/nginx

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update settings.py

erges699 committed:
Update docker-compose-stage.yml

erges699 committed:
update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update docker-compose-stage.yml

erges699 committed:
Update nginx.conf

erges699 committed:
Update docker-compose-stage.yml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 committed:
Update stage_deploy.yaml

erges699 commented:
Победил деплой. Из ветки feature/modify_ci_cd работает.
Установил Docker, mc

deploy:
    name: Deploy changes on server
    runs-on: ubuntu-latest
    environment:
      name: stage_deploy
    needs: build-and-push-image-to-github-packages
    steps:
      - name: Checkout
        uses: actions/checkout@v3
        with:
          ref: feature/modify_ci_cd <- надо изменить на develop
Для теста и настройки деплоя изменил, чтобы копировать infra/docker-compose-stage.yml

erges699 committed:
Update Dockerfile

erges699 committed:
Update development.txt
